ipv-api/src/main/java/uk/gov/di/authentication/ipv/lambda/IPVCallbackHandler.java

```diff
@@ -306,9 +303,55 @@ public APIGatewayProxyResponseEvent handleRequest(
                     AuditService.UNKNOWN,
                     userProfile.getPhoneNumber(),
                     persistentId);
-
-            var userIdentityError = validateUserIdentityResponse(userIdentityUserInfo);
+            var userIdentityError =
+                    ipvCallbackHelper.validateUserIdentityResponse(userIdentityUserInfo);
             if (userIdentityError.isPresent()) {
+                var accountInterventionStatus =
+                        ipvCallbackHelper.getAccountInterventionStatus(internalPairwiseSubjectId);
+                if (configurationService.isAccountInterventionServiceActionEnabled()) {
+                    ipvCallbackHelper.doAccountIntervention(accountInterventionStatus);
+                }
+
+                var returnCode = userIdentityUserInfo.getClaim(RETURN_CODE.getValue());
+                if (returnCode instanceof List<?> returnCodeList && !returnCodeList.isEmpty()) {
+                    var ipAddress = IpAddressHelper.extractIpAddress(input);
+                    if (!isReturnCodePresentAndRequested(clientRegistry, authRequest)) {
+                        LOG.warn("SPOT will not be invoked. Returning Error to RP");
+                        var errorResponse =
+                                new AuthenticationErrorResponse(
+                                        authRequest.getRedirectionURI(),
+                                        OAuth2Error.ACCESS_DENIED,
+                                        authRequest.getState(),
+                                        authRequest.getResponseMode());
+                        return generateApiGatewayProxyResponse(
+                                302,
+                                "",
+                                Map.of(ResponseHeaders.LOCATION, errorResponse.toURI().toString()),
+                                null);
+                    } else {
+                        LOG.info("Generating auth code response");
```

Comment:
This is really nit picky, but I'd argue that we should maybe mention the return code in here, as we've just gone through that step of the logic - just so it's clear for someone checking the logs that we're generating an auth code response in a slightly unhappy state